File: January2022/C++/Minimum_Moves_to_Reach_Target_Score.cpp
class Solution {
public:
    int minMoves(int target, int maxDoubles) {
        int ans=0;
        while(target>1){
            if(target%2){
                target--;
                ans++;
            }else{
                if(maxDoubles){
                    target/=2;
                    ans++;
                    maxDoubles--;                    
                }else{
                    return ans+target-1;
                    target--;
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: Contest/MinimumMovesToReachTargetScore.java


class Solution {


	public int minMoves(int target, int maxDoubles) {
		int count = 0;
		while(target!=1){
			if(target %2 == 1){
				count++;
				target = target-1;
			} else {
				// target is even no
				if(maxDoubles>0){
					target = target /2;
					maxDoubles--;
					count++;

				} else {
					count += target-1;
					target =1;
				}
			}
		}
		return count;



	}
}